### DISCUSSION TITLE
How to say greeting to new Group Users? bot.on is deprecated

### DISCUSSION THREAD

**Discussion by jvillalbaj2lc** [telegraf/telegraf#1777] [Q&A] [answered]
Hi! I'm testing out the module, so far is awesome!!

I'm trying to say hello to all new members of the group but when I write the code says its deprecated.

`bot.on('new_chat_members', (ctx) => {
...
});`

What is the new way to listen for new group members?

Thanks in advance.!

**Comment by MKRhere** [comment, marked as answer]
`bot.on` is not deprecated, but using message subtypes with `bot.on` is. The modern way of doing this is simply:

```TS
import { message } from "telegraf/filters";

bot.on(message("new_chat_members"), handler);
```

Read more about filters here: https://github.com/telegraf/telegraf/releases/tag/v4.11.0

**Reply by jvillalbaj2lc** [reply]
Thank you very much!!!

**Reply by PttRulez** [reply]
When i use this code, bot replies only to messages which start with slash '/'
`bot.on(message('text'), (ctx) => {
	ctx.reply(ctx.message);
});`
![CleanShot 2023-11-13 at 15 36 11@2x](https://github.com/telegraf/telegraf/assets/6817196/ed88ac5a-faff-46d3-98f9-0747ad354b76)

**Reply by MKRhere** [reply]
Go to t.me/Botfather, edit your bot and turn off "Group Privacy". After changing this setting, you need to remove and add the bot back to the group for privacy changes to take effect.

**Reply by PttRulez** [reply]
@MKRhere thank you. Yeah that's what I finally did. But didn't need to remove him from group to take effect.
But the same behavior was also not only in group but in 1-on-1 chat with the bot. Now both in group and 1-on-1 the issue is solved :)
